### PR TITLE
[EPIC] Vouched pools 

### PR DESCRIPTION
**Pull request by 0xlinus**

### CODE REVIEW

File: src/components/stats/List.tsx [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+  const router = useRouter()
+  const { data: voucherEnsAddress, isValidating } = useEnsLookUpAddress(id)
+
+  return (
+    <ButtonPrimaryLightSm
+      disabled={isValidating}
+      onClick={(e) => {
+        e.preventDefault()
+        e.stopPropagation()
+        router.push(`/?voucher=${voucherEnsAddress}`)
```
Comment (saeta-eth) [inline src/components/stats/List.tsx:100]: I think we can use Shallow Routing here. Ref: https://nextjs.org/docs/routing/shallow-routing

Comment (0xlinus) [reply src/components/stats/List.tsx:100]: good call

File: codegen.js [hunk 2]
```diff
@@ -8,8 +8,8 @@ module.exports = {
   overwrite: true,
   schema: [
     // process.env.NEXT_PUBLIC_GRAPH_ENDPOINT_MAINNET,
-    process.env.NEXT_PUBLIC_GRAPH_ENDPOINT_OPTIMISM,
-    // process.env.NEXT_PUBLIC_GRAPH_ENDPOINT_GOERLI,
+    // process.env.NEXT_PUBLIC_GRAPH_ENDPOINT_OPTIMISM,
+    process.env.NEXT_PUBLIC_GRAPH_ENDPOINT_GOERLI,
```
[the remark this answers is not in the document]
Comment (0xlinus) [reply codegen.js:12]: Cant do that. Will break the app. First it needs to be approved. then deploy the (updated) SG and finally update this

[the remark this answers is not in the document]
Comment (saeta-eth) [reply codegen.js:12]: Yeah, this is what I meant, a reminder to change it before going to prod

File: pages/stats.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      <LeftSidebarLayout>
+        <SectionIntro
+          backgroundImage="/resources/svg/bg-sponsors.svg"
+          backgroundPosition="100% 100px"
+          title="Stats"
+        >
+          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt
+          ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation
+          ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in
+          reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.
```
Comment (saeta-eth) [inline pages/stats.tsx:23]: Looks like a placeholder text, do we have the wording?

File: src/hooks/useEnsResolvers.tsx [hunk 4]
```diff
@@ -45,3 +45,26 @@ export const ensResolver = async (name: string) => {
   }
   return name
 }
+
+// Get address by ens name
+export const useEnsResolver = (name: string | undefined) => {
+  const { data, isValidating } = useSWR<string | null>(
+    mainnetRpcProvider && name ? ['ensResolver', name] : null,
+    async () => {
+      if (!name || !isValidENSName) return null
```
Comment (saeta-eth) [inline src/hooks/useEnsResolvers.tsx:54]: Looks like `isValidENSName` is a function

File: src/hooks/aelin/vouched-pools/useVoucherAddress.tsx [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+import { ZERO_ADDRESS } from '@/src/constants/misc'
+
+export default function useVoucherAddress() {
+  const {
+    query: { voucher },
+  } = useRouter()
+
+  const [voucherAddress, setVoucherAddress] = useState<string | null>(null)
+  const { data: aelinVouchAddress } = useEnsResolver(
+    process.env.NEXT_PUBLIC_AELIN_VOUCHER_ENS_ADDRESS as string,
```
Comment (saeta-eth) [inline src/hooks/aelin/vouched-pools/useVoucherAddress.tsx:16]: Would be good to move this env var to `config/env.ts` file

File: src/hooks/aelin/usePoolSupportsMethod.tsx [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+import { useAelinPoolTransaction } from '../contracts/useAelinPoolTransaction'
+import { useAelinPoolUpfrontDealTransaction } from '../contracts/useAelinPoolUpfrontDealTransaction'
+import { ParsedAelinPool } from './useAelinPool'
+import { ZERO_BN } from '@/src/constants/misc'
+import { AelinPool, AelinUpfrontDeal } from '@/types/typechain'
+
+type AelinPoolMethodName = keyof AelinPool['functions']
+type AelinUpFrontDealMethodName = keyof AelinUpfrontDeal['functions']
+
+export function usePoolSupportsMethod(
```
Comment (saeta-eth) [inline src/hooks/aelin/usePoolSupportsMethod.tsx:14]: This is great!

File: pages/stats.tsx (hunk 3 above)
Comment (0xlinus) [reply pages/stats.tsx:23]: yeah good catch. forgot about that one lol

**Review by saeta-eth — APPROVED** [review]

**Pull request merged**